components/dashboard/src/settings/Plans.tsx

```diff
@@ -4,13 +4,461 @@
+                <span>If you are interested in purchasing a plan for a team, purchase a Team plan with one centralized billing. <a className="underline" href="https://www.gitpod.io/docs/teams/" target="_blank">Learn more</a></span>
+            </div>
+            {!!confirmUpgradeToPlan && <Modal visible={true} onClose={() => setConfirmUpgradeToPlan(undefined)}>
+                <h3>Upgrade to {confirmUpgradeToPlan.name}</h3>
+                <div className="border-t border-b border-gray-200 mt-2 -mx-6 px-6 py-2">
+                    <p className="mt-1 mb-4 text-base">You are about to upgrade to {confirmUpgradeToPlan.name}.</p>
+                    {!Plans.isFreePlan(currentPlan.chargebeeId) && <div className="flex rounded-md bg-gray-200 p-4 mb-4">
+                        <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+                        <span>For this billing cycle you will be charged only the total difference ({(confirmUpgradeToPlan.currency === 'EUR' ? '€' : '$') + (confirmUpgradeToPlan.pricePerMonth - applyCoupons(currentPlan, appliedCoupons).pricePerMonth)}). The new total will be effective from the next billing cycle.</span>
+                    </div>}
+                    <div className="flex rounded-md bg-gitpod-kumquat-light p-4 mb-4">
+                        <img className="w-4 h-4 m-1 ml-2 mr-4" src={exclamation} />
+                        <span className="text-red-700">Total: {(confirmUpgradeToPlan.currency === 'EUR' ? '€' : '$') + confirmUpgradeToPlan.pricePerMonth} per month</span>
+                    </div>
+                </div>
+                <div className="flex justify-end mt-6">
+                    <button onClick={doUpgrade}>Upgrade</button>
+                </div>
+            </Modal>}
+            {!!confirmDowngradeToPlan && <Modal visible={true} onClose={() => setConfirmDowngradeToPlan(undefined)}>
+                <h3>Downgrade to {confirmDowngradeToPlan.name}</h3>
+                <div className="border-t border-b border-gray-200 mt-2 -mx-6 px-6 py-2">
+                    <p className="mt-1 mb-4 text-base">You are about to downgrade to {confirmDowngradeToPlan.name}.</p>
+                    <div className="flex rounded-md bg-gray-200 p-4 mb-4">
+                        <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+                        {!Plans.isFreePlan(confirmDowngradeToPlan.chargebeeId)
+                            ? <span>Your account will downgrade to {confirmDowngradeToPlan.name} on the next billing cycle.</span>
+                            : <span>Your account will downgrade to {confirmDowngradeToPlan.name}. The remaining hours in your current plan will be available to use until the next billing cycle.</span>}
+                    </div>
+                </div>
+                <div className="flex justify-end mt-6">
+                    <button className="bg-red-600 border-red-800" onClick={doDowngrade}>Downgrade</button>
```

Comment:
**suggestion**: This could be more descriptive!

```suggestion
                    <button className="bg-red-600 border-red-800" onClick={doDowngrade}>Downgrade Plan</button>
```